packages/firebase_core/firebase_core_web/lib/firebase_core_web.dart

```diff
@@ -30,14 +30,14 @@ FirebaseAppPlatform _createFromJsApp(firebase.App jsApp) {
 /// Returns a [FirebaseOptions] instance from [firebase.FirebaseOptions].
 FirebaseOptions _createFromJsOptions(firebase.FirebaseOptions options) {
   return FirebaseOptions(
-    apiKey: options.apiKey.toDart,
-    authDomain: options.authDomain.toDart,
-    databaseURL: options.databaseURL.toDart,
-    projectId: options.projectId.toDart,
-    storageBucket: options.storageBucket.toDart,
-    messagingSenderId: options.messagingSenderId.toDart,
-    appId: options.appId.toDart,
-    measurementId: options.measurementId.toDart,
+    apiKey: options.apiKey?.toDart ?? '',
+    projectId: options.projectId?.toDart ?? '',
+    authDomain: options.authDomain?.toDart,
+    databaseURL: options.databaseURL?.toDart,
+    storageBucket: options.storageBucket?.toDart,
```

Comment:
I'm also wondering, but since the apiKey for instance can be `undefined` according to [the document](https://firebase.google.com/docs/reference/js/app.firebaseoptions#firebaseoptionsapikey), but we don't allow a null value in the Object in the front-facing api, I can either throw an issue or give a default value

Reply:
ah, I see. I think the default value is fine for apiKey, appId, messagingSenderId, projectId.

But wondering if authDomain, databaseURL, storageBucket should have an empty string default

Reply:
Our front facing api in firebase_core allows null values so I don't think a default value is required. Just for clarification, the object we are creating here is from firebase_core, so I'm just conforming to its type.